index.js

```diff
@@ -0,0 +1,209 @@
+    'lines-between-class-members': 'off',
+    'max-lines': 'error',
+    'no-console': 'error',
+    'no-debugger': 'error',
+    'no-extra-boolean-cast': 'off',
+    'no-plusplus': 'off',
+    'no-restricted-properties': [
+      'error',
+      { object: 'document', property: 'querySelector' },
+      { object: 'document', property: 'querySelectorAll' },
+    ],
+    'no-undef': 'error',
+    'no-underscore-dangle': 'error',
+    'no-unused-expressions': 'off',
+    'react/static-property-placement': 'off',
+    'unicorn/catch-error-name': 'error',
+    'unicorn/custom-error-definition': 'off',
+    'unicorn/error-message': 'error',
+    'unicorn/escape-case': 'error',
+    'unicorn/explicit-length-check': 'error',
+    'unicorn/new-for-builtins': 'error',
+    'unicorn/no-abusive-eslint-disable': 'error',
+    'unicorn/no-array-instanceof': 'error',
+    'unicorn/no-console-spaces': 'error',
+    'unicorn/no-fn-reference-in-iterator': 'off',
+    'unicorn/no-for-loop': 'error',
+    'unicorn/no-hex-escape': 'error',
+    'unicorn/no-new-buffer': 'error',
+    'unicorn/no-process-exit': 'error',
+    'unicorn/no-unreadable-array-destructuring': 'error',
+    'unicorn/no-unsafe-regex': 'off',
+    'unicorn/no-unused-properties': 'error',
+    'unicorn/no-zero-fractions': 'error',
+    'unicorn/number-literal-case': 'error',
+    'unicorn/prefer-exponentiation-operator': 'error',
+    'unicorn/prefer-includes': 'error',
+    'unicorn/prefer-node-remove': 'error',
+    'unicorn/prefer-query-selector': 'error',
+    'unicorn/prefer-starts-ends-with': 'error',
+    'unicorn/prefer-text-content': 'error',
+    'unicorn/prefer-type-error': 'error',
+    'unicorn/prevent-abbreviations': [
+      'error',
+      {
+        replacements: {
+          ref: false,
+          refs: false,
+          prop: false,
+          props: false,
+          src: false,
+          param: false,
+          params: false,
+          args: false,
+        },
+      },
+    ],
+    'unicorn/throw-new-error': 'error',
+  },
+  overrides: [
+    {
+      files: ['*.vue'],
+      parser: 'vue-eslint-parser',
+      rules: {
+        'prettier/prettier': ['off'],
+      },
+    },
+    {
+      files: ['*.ts'],
+      ...typeScriptSettings,
+    },
+    {
+      files: ['*.tsx'],
+      ...typeScriptSettings,
+      extends: [...typeScriptSettings.extends, ...reactSettings.extends],
+      rules: {
+        ...typeScriptSettings.rules,
+        ...reactSettings.rules,
+        'react/prop-types': 'off',
+        '@typescript-eslint/naming-convention': [
+          'error',
+          {
+            selector: 'default',
+            format: ['camelCase'],
+            leadingUnderscore: 'forbid',
+            trailingUnderscore: 'forbid',
+          },
+          {
+            selector: 'typeLike',
+            format: ['PascalCase'],
+          },
+          {
+            selector: 'variable',
+            // Exception for FunctionComponents
```

Comment:
We also need this override for `*.styles.ts/js` when using styled components.

Reply:
I don't think treating styled components in special files makes sense. There shouldn't be any different between a styled component and a "regular" component. Treating those as special only leads to having many one-off components rather than components that are reusable. Think having many StyledWrapper components in a project rather than having a Box component that exposes the props that are typical for wrappers (and other use cases).

Reply:
I favor reusable (styled) components as well, but that doesn't mean that you won't have any page/component specific styles at all. Those will still require specific styled components to be created.

On medium-large size components (splitting up is not always desired), adding even more styled components in the same file, can be come too much. So you want to move them into a separate file.

That file doesn't have any JSX in it (just styled css stuff), thus shouldn't be named jsx/tsx - although we can, which is another solution.

In one of our projects, which is quite large and needs structure, we have a set of dedicated files to use:
* `.styles.ts` - for styled components, all exported as `Styled[ComponentName][StyleName]`
* `.types.ts` - component specific types (for API filters, etc)
* `.utils.ts` - for helper functions related to the component
* `.stories.ts` - for storybook stories
* `.config.ts` - for component specific configuration


I'm not saying this splitting off is something we have to do, and for smaller projects it's perfectly fine to have styled components inside the react component, but we should also support splitting it up.

> Think having many StyledWrapper components in a project rather than having a Box component that exposes the props that are typical for wrappers (and other use cases).

Even then, there are always custom styles that will need to be added. But maybe this whole thing warrants a separate conversation :)